packages/opentelemetry-resources/test/Resource.test.ts

```diff
@@ -253,11 +254,33 @@ describe('Resource', () => {
       await resource.waitForAsyncAttributes?.();
 
       assert.ok(
-        debugStub.calledWithMatch(
-          "a resource's async attributes promise rejected"
-        )
+        debugStub.calledWithMatch('promise rejection for resource attribute')
       );
     });
+
+    it('should guard against asynchronous attribute rejections', async () => {
+      const ee = new EventEmitter();
+      const badAttribute = new Promise<string>((resolve, reject) => {
+        ee.on('fail', reason => reject(reason));
+      });
+      const goodAttribute = new Promise<string>((resolve, reject) => {
+        ee.on('fail', reason => resolve(reason));
+      });
+
+      const res = resourceFromAttributes({ badAttribute, goodAttribute });
+
+      let noUnhandledRejection = true;
+      function onUnhandledRejection() {
+        noUnhandledRejection = false;
+      }
+      process.once('unhandledRejection', onUnhandledRejection);
+      ee.emit('fail', 'resource attribute value promise rejected');
+      // yield to event loop to make sure we don't miss anything
+      await new Promise((resolve, reject) => setTimeout(resolve, 1));
+      assert.ok(noUnhandledRejection);
+      await res.waitForAsyncAttributes?.();
+      assert.notDeepStrictEqual(res.attributes, { goodAttribute: 'fail' });
```

Comment:
Should there be  a `try/finally` around here to `process.removeListener('unhandledRejection', onUnhandledRejection)`? Otherwise this test will possibly have a side-effect on other tests in the same process.

Reply:
probably